Scripts/summaryStatistics/biophysics/growth_rate_analysis.py

```diff
@@ -0,0 +1,247 @@
+    """
+    
+    kneedle = KneeLocator(x, y, S=1.0, curve=curvature_type, direction=direction_type)
+    knee = kneedle.knee
+    
+    return knee
+    
+def fit_line(x, y):
+    """
+    Fit a line to [x, y] data. Get regression coefficients and R^2.
+    
+    @param  x           (1D nparray) X data
+    @param  y           (1D nparray) Y data
+    @return slope       Slope of linear fit
+    @return intercept   Intercept of linear fit
+    @return r_squared   R^2 of linear fit
+    """
+    # Necessary to reshape data for LinearRegression model
+    x_reshape = x.reshape((-1,1))
+    
+    # Linear regression
+    model = LinearRegression().fit(x_reshape, y)
+    r_squared = model.score(x_reshape, y)
+    model_prediction = model.predict(x_reshape)
+    slope = model.coef_[0] # slope
+    intercept = model.intercept_ # intercept
+    r_squared = model.score(x_reshape, y)
+    
+    return slope, intercept, r_squared
+    
+def main(cells, dt, bin_radius=4):
+    """
+    The main function.
+    Calculates the slope of growth_rate vs. radial cell position in the colony in growth portions of the colony.
+    This function is to be called by pyabc.
+    
+    @param  dt          Time step
+    @param  bin_radius  Width to perform radial average in um (int)
+    @param  cells       cellStates dict
+    @return slope       Slope of growth_rate vs. position in colony
+    """
+    # Get colony centroid
+    centerX, centerY = get_colony_centroid(cells)
+    
+    # Filter cells by age to exclude cells that were just born
+    cells_filtered = filter_by_cell_age(cells, min_age=2)
+    
+    # Initialize storage variables
+    r = np.zeros(len(cells_filtered))
+    growth_rate = np.zeros(len(cells_filtered))
+    
+    # Collect radial positions and growth rates
+    for i, cell in enumerate(cells_filtered.values()):
+        r[i] = get_radial_position(cell, centerX, centerY)
+        growth_rate[i] = cell.strainRate_rolling/dt #cell.normalized_growth_rate
+        if math.isnan(growth_rate[i]):
+            growth_rate[i] = 0
+    
+    # Create a dataFrame with each cell placed in a bin corresponding to radial position
+    df = bin_by_radius(r, growth_rate, bin_radius, centerX, centerY)
+    
+    # Calculate statistics of binned measurements
+    mean_df, std_df = get_binned_statistics(df)   
+    
+    # Get bin ranges for plotting purposes
+    bin_ranges = define_bin_ranges(bin_radius, r)
+        
+    # Find knee, then get range of points to fit line
+    knee = find_knee(bin_ranges, mean_df['Growth Rate'])
+    try:
+        knee_index = np.where(bin_ranges == knee)[0][0] 
+    except:
+        print("No knee, fit to entire curve")
+        knee_index = 0
+    bin_ranges_linear = bin_ranges[knee_index:]
+    growth_rate_linear = mean_df['Growth Rate'].to_numpy()[knee_index:]
+    
+    # Fit line
+    slope, intercept, r_squared = fit_line(bin_ranges_linear, growth_rate_linear)
+    
+    return slope
+
+if __name__ == "__main__":
```

Comment:
It is always better and more professional to write your test code in a separate file and not in the same file as the main code of your script.